### PR TITLE
[G-API] Support GFrame for infer

### PR DESCRIPTION
**Pull request by TolyaTalamanov**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

```
force_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

Xbuild_image:Custom=centos:7
Xbuildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

build_image:Custom=ubuntu-openvino-2021.1.0:20.04
build_image:Custom Win=openvino-2021.1.0
build_image:Custom Mac=openvino-2021.1.0

test_modules:Custom=gapi
test_modules:Custom Win=gapi
test_modules:Custom Mac=gapi

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

File: modules/gapi/src/backends/ie/giebackend.cpp
Comment (dmatveev) [inline modules/gapi/src/backends/ie/giebackend.cpp:466]: So do you use it really?

**Review by dmatveev — APPROVED** [review]
👍  approved, thanks!

File: modules/gapi/src/backends/ie/giebackend.cpp
Comment (TolyaTalamanov) [reply modules/gapi/src/backends/ie/giebackend.cpp:466]: use it to distinguish `cv::GMat` & `cv::GFrame`

**Comment by TolyaTalamanov** [comment]
@alalek CI is read: `error C2039: 'NV12Blob': is not a member of 'InferenceEngine' `
Log: https://pullrequest.opencv.org/buildbot/builders/precommit_custom_windows/builds/1707/steps/compile%20release/logs/stdio

Can it be the problem with OpenVINO version ? 

https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/samples/hello_nv12_input_classification/main.cpp#L157

**Comment by alalek** [comment]
Mentioned sample is available since 2019 R2.

You are testing using OV LTS release:

> openvino-2020.3.0

Need to add proper version check through `#if` if this is necessary.

BTW, You can use different OV versions on Linux/Windows/Mac configurations to perform robust checking (recommended).

**Comment by TolyaTalamanov** [comment]
@alalek Custom jobs failed due known problems (cpu options isn't supported in this OV version)

I think it can be merged

**Comment by alalek** [comment]
> Custom jobs failed due known problems (cpu options isn't supported in this OV version)
> I think it can be merged

Really? I see this:

> Results:
> Failed compile release

and missing build images (OV LTS 2020.3.1 is not supported for now). Consider using 2021.1 instead.

---

Compilation must be fixed before merge.

**Comment by TolyaTalamanov** [comment]
> > Custom jobs failed due known problems (cpu options isn't supported in this OV version)
> > I think it can be merged
> 
> Really? I see this:
> 
> > Results:
> > Failed compile release
> 
> and missing build images (OV LTS 2020.3.1 is not supported for now). Consider using 2021.1 instead.
> 
> Compilation must be fixed before merge.

2021.1 isn't supported on CI, is it ?

**Comment by alalek** [comment]
> 2021.1 isn't supported on CI, is it ?

Supported (on Linux in Ubuntu 20.04 container): https://github.com/opencv/opencv/wiki/CI-configuration

File: modules/gapi/src/backends/ie/giebackend.cpp
Comment (mpashchenkov) [inline modules/gapi/src/backends/ie/giebackend.cpp:785]: Isn't this checked in `switch`? For what case is it, if not.

**Review by smirnov-alexey — APPROVED** [review]

**Comment by smirnov-alexey** [comment]
Was able to successfully built with both 2020.03 and 2021.01

**Comment by smirnov-alexey** [comment]
@alalek can you have a look at this https://pullrequest.opencv.org/buildbot/builders/precommit_custom_linux/builds/5124?

**Comment by alalek** [comment]
> openvino-2021.1

Use this Wiki page for proper names of available build images: https://github.com/opencv/opencv/wiki/CI-configuration

Should be `*openvino-2021.1.0*`

**Comment by smirnov-alexey** [comment]
> > openvino-2021.1
> 
> Use this Wiki page for proper names of available build images: https://github.com/opencv/opencv/wiki/CI-configuration
> 
> Should be `*openvino-2021.1.0*`

Missed that, thanks!

**Comment by smirnov-alexey** [comment]
@alalek can we merge this if the build will be green? There was a warning on Mac which hopefully will be fixed and other warnings are:
```
CMake Warning at /opt/intel/openvino_2021.1.110/deployment_tools/inference_engine/share/ie_parallel.cmake:6 (find_package):
  By not providing "FindTBB.cmake" in CMAKE_MODULE_PATH this project has
```
and
```
CMake Warning:
  Manually-specified variables were not used by the project:

    ENABLE_CXX11
    INF_ENGINE_RELEASE
    WITH_INF_ENGINE
```

**Comment by alalek** [comment]
Yes, even with warnings on "optional" configuration (in scope of OpenCV, not G-API) if these warnings are not related to this PR (the same issues are with other G-API PRs on the same configurations).

**Comment by dmatveev** [comment]
Let's merge this - we need this feature for the next internal update

**Comment by smirnov-alexey** [comment]
@alalek the build is green except for 2 warnings I've mentioned above (which have nothing to do with this PR). Can we merge this?

**Comment by smirnov-alexey** [comment]
@alalek Thanks!

**Comment by dmatveev** [comment]
Fantastic!

**Pull request merged**
